baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/safety/SafetyAnnotations.java

```diff
@@ -122,6 +124,30 @@ public static Safety getSafety(@Nullable Type type, VisitorState state) {
         return Safety.UNKNOWN;
     }
 
+    public static Safety getDirectSafety(@Nullable Symbol symbol, VisitorState state) {
```

Comment:
Extracted to avoid duplicating effort between `ASTHelpers.hasAnnotation` and our annotation search. `ASTHelpers.hasAnnotation` searches supertypes, but not their interfaces, and there's no sense doing it twice.

I took the opportunity to use an error-prone supplier for names, which should reduce overhead a bit.